content/blog/bare-metal-provisioning-on-hpe-greenlake-using-terraform.md

```diff
@@ -0,0 +1,353 @@
+
+***Note: Compute group is an AKA Project.***
+
+1. Navigate to HPE GreenLake for Private Cloud Services card -> Bare Metal -> Compute Groups.
+
+![BMaaS Compute Groups](/img/compute_group_list.png "BMaaS Compute Groups")
+
+2. Click on the desired compute group and then extract the ID from the browser URL seen at that time.  
+
+   This will be exported in the environment variable **HPE_METAL_PROJECT_ID** in the later section.
+
+![Compute group ID](/img/compute_group_id.png "Compute group ID")
+
+## Install Terraform
+
+Next, get your system ready to run Terraform. In case this has not been done yet:
+
+1. Download and install Terraform, version v0.13 or later.
+   For more information, see [https://learn.hashicorp.com/tutorials/terraform/install-cli](<1. https://learn.hashicorp.com/tutorials/terraform/install-cli>). 
+2. Verify the installation with **terraform -help.** 
+
+   At this point, you are ready to start building your infrastructure description file. 
+
+# Deploy Compute Instance
+
+## Select Terraform provider with bare metal service configurations
+
+1. **Export the following environment variables on your setup.**
+
+   E﻿xport the Tenant ID:
+
+   ```shell
+   export HPEGL_TENANT_ID="<Tenant ID>
+   ```
+
+   Export the API Client credentials that you obtained when you created an API Client within HPE GreenLake Central:
+
+   ```shell
+   export HPEGL_USER_ID="<API Client ID>"
+   export HPEGL_USER_SECRET="<API Client Secret>"
+   export HPEGL_IAM_SERVICE_URL="<Issuer URL>"
+   ```
+
+   E﻿xport the Compute group ID:
+
+   ```
+   # compute group/project ID
+   export HPEGL_METAL_PROJECT_ID="<Compute group ID>"
+   ```
+
+    ﻿Export bare metal service REST URL:
+
+   ```
+   # Production Environment:  https://client.greenlake.hpe.com/api/metal
+   # Integration Environment: https://client.greenlake.hpe-gl-intg.com/api/metal
+   # local development: http://localhost:3002
+
+   export HPEGL_METAL_REST_URL="<Metal Service REST Base URL>"
+   ```
+2. **Configure the Terraform provider.**
+
+   Create an empty folder and put a file in it called main.tf with the following contents:
+
+   **main.tf:**
+
+   ```hcl
+   terraform {
+     required_providers {
+       hpegl = {
+         source  = "HPE/hpegl"
+         version = ">= 0.3.12"
+       }
+     }
+   }
+
+   provider "hpegl" {    
+     # metal block for configuring bare metal resources.
+     metal {   
+     } 
+   }
+   ```
+
+   T﻿his tells Terraform that you are going to be using HPE/hpegl as your provider and you are using the bare metal service.
+
+## Write resource configuration for compute instance creation
+
+To deploy compute instance, you need to use the **hpegl_metal_host** terraform resource.
```

Comment:
Escape the "underscore" character for the variable "hpegl_metal_host" throughout the post.